### DISCUSSION TITLE
Really having hard time reading and understanding documentation?  Book Suggestions

### DISCUSSION THREAD

**Discussion by jiaulislam** [sqlalchemy/sqlalchemy#8505] [Usage Questions]
Hi,  Sorry if i am being dumb but i am really having hard time to understand the documentation of SQL alchemy. I know its just me. But is there any book anyone suggests that can help me to go through it. Many terms even i Don't understand what's that even mean(e.g: declarative base arguments,  Construction of table in declarative way but what type of custom arguments is there in class) .  So i guess i really work in these area,  So i would be grateful is there any book to learn SQL Alchemy. 

Thanks

**Comment by zzzeek** [comment]
the only book that was published is this one:  https://www.oreilly.com/library/view/essential-sqlalchemy-2nd/9781491916544/

it's going to be a little bit dated now but certainly if it helps explain general things, the general idea of things hasn't changed all that much.

Also, I would recommend videos on youtube, including my own at https://www.youtube.com/watch?v=1Va493SMTcY&t=9s , which you can do along with a code tutorial, but there's lots of others on youtube if you search.

**Reply by zzzeek** [reply]
also for the SQLAlchemy docs itself, you'd want to do the [quickstart](https://docs.sqlalchemy.org/en/14/orm/quickstart.html), play around with it, then the [tutorial](https://docs.sqlalchemy.org/en/14/tutorial/index.html), which is long, but you can run the code examples and just spend time familiarizing with each step.  it starts very minimally.     but i would advise playing with small code examples for real.

**Reply by techUdayMungalpara** [reply]
i also face the same issue try to learn sql alchemy can you share way to learn and any resource Thanks.

**Reply by jiaulislam** [reply]
> i also face the same issue try to learn sql alchemy can you share way to learn and any resource Thanks.

To this day I couldn't find any new books to study. But I learned now by trying and reading docs in Jupyter Notebooks and seeing the outputs. It helped me a lot.

**Reply by techUdayMungalpara** [reply]
can you share code on github Thanks .

**Comment by jiaulislam** [comment]
> can you share code on github Thanks .

well there's not much of code. But for example you can see my [Repo](https://github.com/jiaulislam/notebook-playground)
